CONTRIBUTING.md

```diff
@@ -2,7 +2,156 @@
 
 We welcome all contributions, bug reports, and suggestions!
 
-## Installing the development version
+* Ask support and usage questions in [Discussions](https://github.com/koordinates/kart/discussions)
+* Read and submit bug reports or feature requests at [Issues](https://github.com/koordinates/kart/issues)
+
+We're moving to CMake as a better system for building Kart and its
+dependencies. Currently CMake is still a work in progress and doesn't yet build
+all the vendor dependencies or create packages suitable for distribution.
+
+## Building the development version with CMake (macOS & Linux)
+
+Requirements for macOS & Linux:
+* CMake >= v3.21
+* GDAL >= v3.3.2
+* Git >= v2.31
+* LibGit2 >= v1.1.0
+* OpenSSL >= v1.1
+* PostgreSQL client library (libpq)
+* Python >= 3.7
+* Spatialindex >= v1.9.3
+* SpatiaLite >= v5.0.1
+* SQLite3 >= v3.31.1
+* SWIG
+* unixODBC >= v2.3.9 (macOS/Linux only)
+
+Clone Kart from Github:
+
+```console
+$ git clone https://github.com/koordinates/kart.git
+$ cd kart
+```
+
+### Installing dependencies on macOS
+
+If you're a Homebrew user, you can get all of those via:
+
+```console
+$ brew install --upgrade cmake gdal git openssl@1.1 libpq python@3.7 \
+    spatialindex libspatialite sqlite3 swig unixodbc
+```
+
+Then configure Kart:
+```
+$ cmake -B build -S .
```

Comment:
I think the MacOS instructions are missing the bit about installing a custom libgit2 (it's below in the Linux section, probably just copy/move it):

```
  Could NOT find LibGit2 (missing: LibGit2_LIBRARY LibGit2_INCLUDE_DIR)
  (Required is at least version "1.3.0")
```

If I additionally `brew install libgit2` and try again I get this:

```
-- Found LibGit2: /usr/local/lib/libgit2.dylib (found suitable version "1.3.0", minimum required is "1.3.0")
CMake Error at vendor/CMakeLists.txt:76 (message):
  Found LibGit2 v1.3.0, but git_error_subcode isn't defined.  Build & install
  from https://github.com/koordinates/libgit2/tree/kx-latest
```

Reply:
actually I get that latter error _even if_ I follow the linux install instructions for our custom libgit2. So something else missing I guess

Reply:
> I think the MacOS instructions are missing the bit about installing a custom libgit2 (it's below in the Linux section, probably just copy/move it):

Ah, yeah, decided it was better not to use the Homebrew one in the end.

> If I additionally brew install libgit2 and try again I get this:

Which is expected

> actually I get that latter error even if I follow the linux install instructions for our custom libgit2. So something else missing I guess

1. Did you `brew unlink libgit2`?
2. Does deleting `build/CMakeCache.txt` help? If it found something CMake caches it, so it'll reuse the cache instead of finding it again